modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -541,6 +541,38 @@ def reindex(self, axis, labels, **kwargs):
         return self.__constructor__(new_modin_frame)
 
     def reset_index(self, **kwargs):
+        if self.lazy_execution:
```

Comment:
This is false for distributed engines. What are we trying to speed up?

Reply:
no, it is conditionally True or False depending on whether the frame has delayed computations:
https://github.com/modin-project/modin/blob/4f78260d91042497b5d40305c3647bb00c482b14/modin/core/storage_formats/pandas/query_compiler.py#L254-L266

Here we want to delegate the logic of error-checking to the kernel function in case there's not enough metadata to do so in the main process

Reply:
What logic of error-checking do you mean? I don't see anything related in the kernel.

Reply:
the kernel does this natively since we're using pandas itself (whose `.reset_index` implementation contains the logic), the difference here that in case of lazy execution we'll get an error at the time we'll materialize the results (which seems logical for lazy execution)